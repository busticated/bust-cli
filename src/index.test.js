const version = require('../package.json').version;
const mockTerm = require('./_test/mock-terminal');
const bin = require.resolve('./');


describe('bust cli', function(){
    it('complains about missing command', async function(){
        const { stdout, stderr, code } = await mockTerm(`${bin}`);

        expect(code).toBe(1);
        expect(stdout).toBe('');
        expect(stderr).toMatchSnapshot();
    });

    it('complains about unrecognized command', async function(){
        const { stdout, stderr, code } = await mockTerm(`${bin} wat-nope-nah`);

        expect(code).toBe(1);
        expect(stdout).toBe('');
        expect(stderr).toMatchSnapshot();
    });

    it('prints version', async function(){
        const { stdout, stderr, code } = await mockTerm(`${bin} -v`);

        expect(code).toBe(0);
        expect(stdout).toBe(`${version}\n`);
        expect(stderr).toBe('');
    });

    it('prints help', async function(){
        const { stdout, stderr, code } = await mockTerm(`${bin} help`);

        expect(code).toBe(0);
        expect(stdout).toMatchSnapshot();
        expect(stderr).toBe('');
    });
});
