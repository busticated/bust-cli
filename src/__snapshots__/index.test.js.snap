// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bust cli complains about missing command 1`] = `
"Not enough non-option arguments: got 0, need at least 1

whoops, something went wrong! run with --help
"
`;

exports[`bust cli complains about unrecognized command 1`] = `
"Unknown argument: wat-nope-nah

whoops, something went wrong! run with --help
"
`;

exports[`bust cli prints help 1`] = `
"                                          __                       __
                                         /\\\\ \\\\                     /\\\\ \\\\__
    __   __   __   __   __   __   __   __\\\\ \\\\ \\\\____  __  __    ____\\\\ \\\\ ,_\\\\
   /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\\\\\ \\\\ '__\`\\\\/\\\\ \\\\/\\\\ \\\\  /',__\\\\\\\\ \\\\ \\\\/
 __\\\\/_/_\\\\/_/_\\\\/_/_\\\\/_/_\\\\/_/_\\\\/_/_\\\\/_/_\\\\/_/_\\\\ \\\\ \\\\L\\\\ \\\\ \\\\ \\\\_\\\\ \\\\/\\\\__, \`\\\\\\\\ \\\\ \\\\_  __
/\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\ /\\\\_\\\\\\\\ \\\\_,__/\\\\ \\\\____/\\\\/\\\\____/ \\\\ \\\\__\\\\/\\\\_\\\\
\\\\/_/ \\\\/_/ \\\\/_/ \\\\/_/ \\\\/_/ \\\\/_/ \\\\/_/ \\\\/_/ \\\\/_/ \\\\/___/  \\\\/___/  \\\\/___/   \\\\/__/\\\\/_/


Usage: bust <command> [options]

Commands:
  gitignore  create .gitignore file                                [aliases: gi]

Options:
  --version, -v  display version information                           [boolean]
  --help, -h     display help message                                  [boolean]

"
`;
